WordPress/src/main/java/org/wordpress/android/ui/domains/management/purchasedomain/PurchaseDomainViewModel.kt

```diff
@@ -30,28 +34,61 @@ class PurchaseDomainViewModel @AssistedInject constructor(
 
     fun onNewDomainSelected() {
         analyticsTracker.track(Stat.DOMAIN_MANAGEMENT_PURCHASE_DOMAIN_SCREEN_NEW_DOMAIN_TAPPED)
-        launch {
-            _actionEvents.emit(ActionEvent.GoToDomainPurchasing(domain = domain))
-        }
+        createCart(null, productId, domain, privacy)
     }
 
     fun onExistingSiteSelected() {
         analyticsTracker.track(Stat.DOMAIN_MANAGEMENT_PURCHASE_DOMAIN_SCREEN_EXISTING_SITE_TAPPED)
         launch {
-            _actionEvents.emit(ActionEvent.GoToExistingSite(domain = domain))
+            _actionEvents.emit(ActionEvent.GoToSitePicker(domain = domain))
         }
     }
 
+    fun onSiteChosen(site: SiteModel?) {
+        analyticsTracker.track(Stat.DOMAIN_MANAGEMENT_PURCHASE_DOMAIN_SCREEN_EXISTING_SITE_CHOSEN)
+        createCart(site, productId, domain, privacy)
+    }
+
     fun onBackPressed() {
         launch {
             _actionEvents.emit(ActionEvent.GoBack)
         }
     }
 
+    @Suppress("UNUSED_PARAMETER")
+    fun onDomainRegistrationComplete(event: DomainRegistrationCompletedEvent?) {
+        // TODO Handle domain registration complete
```

Comment:
I've created a separate issue to handle this to keep this PR simple https://github.com/wordpress-mobile/WordPress-Android/issues/19544